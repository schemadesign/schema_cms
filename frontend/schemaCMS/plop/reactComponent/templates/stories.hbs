import React from 'react';
import { storiesOf } from '@storybook/react';

import { {{ pascalCase name }} } from './{{ camelCase name }}.component';

export const defaultProps = {};

storiesOf('{{ pascalCase name }}', module).add('Default', () => <{{ pascalCase name }} {...defaultProps} />);
