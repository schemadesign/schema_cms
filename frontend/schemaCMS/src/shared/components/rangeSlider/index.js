export { RangeSlider } from './rangeSlider.component';
