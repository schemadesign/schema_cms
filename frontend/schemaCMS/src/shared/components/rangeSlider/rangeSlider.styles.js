import styled, { css } from 'styled-components';

const SLIDE_SIZE = '24px';

export const Container = styled.div`
  height: ${SLIDE_SIZE};
  min-width: 100%;
  position: relative;
  display: flex;
  align-items: center;
`;

export const Slider = styled.div`
  height: 3px;
  background: #1d1d20;
  width: 100%;
`;

export const SelectedSlider = styled.div`
  height: 3px;
  background: #71737e;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  width: ${({ size }) => `${size}%`};
  left: ${({ left }) => `${left}%`};
`;

export const slideStyles = css`
  width: ${SLIDE_SIZE};
  height: ${SLIDE_SIZE};
  background: #fff;
  border-radius: 50%;
  cursor: pointer;
  pointer-events: all;
`;

export const sliderStyles = css`
  appearance: none;
  width: 100%;
  height: ${SLIDE_SIZE};
  background: transparent;
  outline: none;
  pointer-events: none;
  margin: 0;
  position: absolute;
  top: 0;
  left: 0;

  &::-webkit-slider-thumb {
    appearance: none;
    ${slideStyles};
  }

  &::-moz-range-thumb {
    ${slideStyles};
  }
`;
export const Min = styled.input`
  ${sliderStyles};
`;
export const Max = styled.input`
  ${sliderStyles};
`;
