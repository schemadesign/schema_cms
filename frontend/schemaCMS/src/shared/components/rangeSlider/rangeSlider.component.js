import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Min, Max, Container, Slider, SelectedSlider } from './rangeSlider.styles';

export class RangeSlider extends PureComponent {
  static propTypes = {
    minValue: PropTypes.number.isRequired,
    maxValue: PropTypes.number.isRequired,
    idMin: PropTypes.string.isRequired,
    idMax: PropTypes.string.isRequired,
    min: PropTypes.number,
    max: PropTypes.number,
    onChange: PropTypes.func.isRequired,
  };

  static defaultProps = {
    min: 0,
    max: 100,
  };

  render() {
    const { min, max, minValue, maxValue, onChange, idMin, idMax } = this.props;
    const min1 = parseInt(min % 1 ? min - 1 : min, 10);
    const max1 = parseInt(max % 1 ? max + 1 : max, 10);
    const mapBetween = (currentNum, minAllowed, maxAllowed, min, max) =>
      ((maxAllowed - minAllowed) * (currentNum - min)) / (max - min) + minAllowed;

    const left = mapBetween(minValue, 0, 100, min1, max1);
    const right = mapBetween(maxValue, 0, 100, min1, max1);
    const width = right - left;

    return (
      <Container>
        <Slider />
        <SelectedSlider left={left} size={width} />
        <Min min={min1} max={max1} value={minValue} type="range" id={idMin} name={idMin} onChange={onChange} />
        <Max min={min1} max={max1} value={maxValue} type="range" id={idMax} name={idMax} onChange={onChange} />
      </Container>
    );
  }
}
