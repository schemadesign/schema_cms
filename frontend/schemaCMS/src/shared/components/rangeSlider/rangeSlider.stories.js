import React, { PureComponent } from 'react';
import { storiesOf } from '@storybook/react';

import { RangeSlider } from './rangeSlider.component';
import { withTheme } from '../../../.storybook/decorators';

export const defaultProps = {
  min: 0,
  max: 100,
  minValue: 10,
  maxValue: 40,
  idMin: 'idMin',
  idMax: 'idMax',
  onChange: Function.prototype,
};

class TestComponent extends PureComponent {
  state = {
    minValue: 10,
    maxValue: 40,
  };

  handleChange = ({ target }) => this.setState({ maxValue: target.value });

  render() {
    return <RangeSlider {...defaultProps} {...this.state} onChange={this.handleChange} />;
  }
}

storiesOf('RangeSlider', module)
  .addDecorator(withTheme())
  .add('Default', () => <TestComponent {...defaultProps} />);
