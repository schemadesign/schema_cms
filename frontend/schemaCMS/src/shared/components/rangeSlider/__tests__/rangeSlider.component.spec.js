import React from 'react';
import { shallow } from 'enzyme';

import { RangeSlider } from '../rangeSlider.component';
import { defaultProps } from '../rangeSlider.stories';

describe('RangeSlider: Component', () => {
  const component = props => <RangeSlider {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });
});
