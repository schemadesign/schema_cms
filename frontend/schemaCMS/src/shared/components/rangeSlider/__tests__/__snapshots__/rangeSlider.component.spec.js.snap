// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RangeSlider: Component should render correctly 1`] = `
<styled.div>
  <styled.div />
  <styled.div
    left={10}
    size={30}
  />
  <styled.input
    id="idMin"
    max={100}
    min={0}
    name="idMin"
    onChange={[Function]}
    type="range"
    value={10}
  />
  <styled.input
    id="idMax"
    max={100}
    min={0}
    name="idMax"
    onChange={[Function]}
    type="range"
    value={40}
  />
</styled.div>
`;
